### ISSUE TITLE
[Machine Learning] Hyperparameter Tuning of Machine Learning Model in Python

### ISSUE THREAD

**Issue by willyngashu** [section-engineering-education/engineering-education#5009]
#### Proposed title of article
[Machine Learning] Hyperparameter Tuning of Machine Learning Model in Python

#### Proposed article introduction
Hyperparameters are parameters that can be fine-tuned and adjusted in order to improve the performance of a machine learning model. Various machine algorithms such as RandomForest and K-nearest neighbor algorithms have parameters that can be adjusted to achieve an optimized model.

In the process of hyperparameter tuning, we create a list of all possible values for hyperparameters.  We then iterate through the list trying to find a random combination of hyperparameters, record the performance of each parameter, and finally return the combination of hyperparameters that will provide an optimal model with the optimal solution.

Finding and tuning the right hyperparameters for any machine learning model is a very crucial aspect of today's development. We have to reduce the model loss function and in the end, we have a better model. We have various techniques for hyperparameter tuning: Random Search, Grid Search, Bayesian Optimization, and Tree-structured Parzen estimators (TPE).

#### Key takeaways 
1. What is hyperparameter tuning.
2. Various techniques and methods used for hyperparameter tuning.
3. Building a machine learning model using RandomForest.
4. Fine-tuning RandomForest parameters.
5. Implementing various techniques and methods for hyperparameter tuning.

#### Article quality
This tutorial is a practical guide, explaining everything from scratch with a step-by-step guide that is easier to follow. This tutorial also uses two approaches in building the model, In the First approach we build a model without tuning and in the second approach, we will fine-tune the parameters to increase the model accuracy.
The tutorial will also discuss the methods for hyperparameter tuning in detail.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by lalith1403** [comment]
Great topic 🚀 , make sure it matches the following: 
* Adds value beyond what is in the official docs and what is covered in other blog sites.
* To avoid potential article closure, it should not overlap with existing EngEd articles or incoming topic suggestions.

Please reference any relevant EngEd articles in yours and build a unique project - Approved.
